core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -128,41 +131,60 @@ private[spark] class CompressedMapStatus(
  * @param numNonEmptyBlocks the number of non-empty blocks
  * @param emptyBlocks a bitmap tracking which blocks are empty
  * @param avgSize average size of the non-empty blocks
+ * @param hugeBlockSizes sizes of huge blocks by their reduceId.
  */
 private[spark] class HighlyCompressedMapStatus private (
     private[this] var loc: BlockManagerId,
     private[this] var numNonEmptyBlocks: Int,
     private[this] var emptyBlocks: RoaringBitmap,
-    private[this] var avgSize: Long)
+    private[this] var avgSize: Long,
+    @transient var hugeBlockSizes: Map[Int, Byte])
```

Comment:
@mridulm 
I removed the `hugeBlockSizesArray` and put `hugeBlockSizes` as transient in constructor. How about this now? I'm not sure how to make this as private. Should I move `hugeBlockSizes` from constructor into Class and create setter for it?